lib/gs1/extensions/check_digit_validation.rb

```diff
@@ -0,0 +1,17 @@
+module GS1
+  module Extensions
+    # Ensures correct check digit validation.
+    #
+    module CheckDigitValidation
+      def valid_check_digit?
+        CheckDigit.with_sequence(data[0..-2]) == data
```

Comment:
Contains false positives FYI. Maybe it is unavoidable.

Reply:
Related to your other comment?